biogtr/datasets/base_dataset.py

```diff
@@ -0,0 +1,145 @@
+"""Module containing logic for loading datasets."""
+from biogtr.datasets import data_utils
+from torch.utils.data import Dataset
+from typing import List
+import torch
+
+
+class BaseDataset(Dataset):
+    """Base Dataset for microscopy and sleap datasets to override."""
+
+    def __init__(
+        self,
+        files: list[str],
+        padding: int,
+        crop_size: int,
+        chunk: bool,
+        clip_length: int,
+        mode: str,
+        augmentations: dict = None,
+        gt_list: str = None,
+    ):
+        """Initialize Dataset.
+
+        Args:
+            files: a list of files, file types are combined in subclasses
+            padding: amount of padding around object crops
+            crop_size: the size of the object crops
+            chunk: whether or not to chunk the dataset into batches
+            clip_length: the number of frames in each chunk
+            mode: `train` or `val`. Determines whether this dataset is used for
+                training or validation. Currently doesn't affect dataset logic
+            augmentations: An optional dict mapping augmentations to parameters.
+                See subclasses for details.
+            gt_list: An optional path to .txt file containing ground truth for
+                cell tracking challenge datasets.
+        """
+        self.files = files
+        self.padding = padding
+        self.crop_size = crop_size
+        self.chunk = chunk
```

Comment:
Add a default for `self.chunks` so we don't get weird attribute missing error if `self.create_chunks()` hasn't been called yet?